cmd/snap-confine/system-shutdown.c

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include <stdbool.h>		// bools
+#include <stdarg.h>		// va_*
+#include <sys/mount.h>		// umount
+#include <sys/stat.h>		// mkdir
+#include <unistd.h>		// getpid, close
+#include <stdlib.h>		// exit
+#include <stdio.h>		// fprintf, stderr
+#include <sys/ioctl.h>		// ioctl
+#include <linux/loop.h>		// LOOP_CLR_FD
+#include <sys/reboot.h>		// reboot, RB_*
+#include <fcntl.h>		// open
+#include <errno.h>		// errno, sys_errlist
+
+#include "system-shutdown-utils.h"
+
+int main(int argc, char *argv[])
+{
+	errno = 0;
+	if (getpid() != 1) {
```

Comment:
A small comment would be great, something like `systemd will *exec()* the shutdown binary` or something. When reading this I was thinking it means `getppid()`. Or maybe just moving the comment below that explains it all a bit up :)